nemo/collections/tts/models/magpietts.py

```diff
@@ -1570,163 +1702,341 @@ def embed_context_text(self, context_text_tokens):
+
+        Returns:
+            Tuple of (tensor_a, tensor_b) both with shape (B, max(T_a, T_b), E).
+        """
+        len_a, len_b = tensor_a.size(1), tensor_b.size(1)
+        if len_a < len_b:
+            padding = torch.zeros(
+                tensor_a.size(0), len_b - len_a, tensor_a.size(2), device=tensor_a.device, dtype=tensor_a.dtype
+            )
+            tensor_a = torch.cat([tensor_a, padding], dim=1)
+        elif len_a > len_b:
+            padding = torch.zeros(
+                tensor_b.size(0), len_a - len_b, tensor_b.size(2), device=tensor_b.device, dtype=tensor_b.dtype
+            )
+            tensor_b = torch.cat([tensor_b, padding], dim=1)
+        return tensor_a, tensor_b
+
+    def _get_context_embeddings(
+        self,
+        batch: Dict[str, torch.Tensor],
+        context_audio_codes: torch.Tensor,
+        context_audio_codes_lens: torch.Tensor,
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Get context embeddings, handling text conditioning if enabled.
+
+        Args:
+            batch: Batch dictionary containing context tokens if text conditioning is used.
+            context_audio_codes: Context audio codes. Shape: (B, C, T_ctx).
+            context_audio_codes_lens: Length of context audio codes. Shape: (B,).
+
+        Returns:
+            Tuple of (context_embedded, context_lens) where:
+                context_embedded: Combined context embedding. Shape: (B, T, E).
+                context_lens: Length of context sequences. Shape: (B,).
+        """
+        context_audio_embedded = self.embed_audio_tokens(context_audio_codes)  # (B, T/frame_stacking, E)
+
+        if not self.use_text_conditioning_encoder:
+            context_lens = torch.ceil(context_audio_codes_lens / self.frame_stacking_factor).to(
+                context_audio_codes_lens.dtype
+            )
+            return context_audio_embedded, context_lens
+
+        # Text conditioning path
+        context_text_tokens = batch['context_text_tokens']
+        context_text_lens = batch['context_text_tokens_lens']
+        context_text_embedded = self.embed_context_text(context_text_tokens)  # (B, L, E)
 
-            if self.model_type == 'multi_encoder_context_tts':
+        # Pad tensors to match sequence lengths
+        context_audio_embedded, context_text_embedded = self._pad_tensors_to_match(
+            context_audio_embedded, context_text_embedded
+        )
+
+        # Interpolate between audio and text embeddings based on has_text_context flag
+        has_text_context = batch['has_text_context'].unsqueeze(-1).unsqueeze(-1).float()  # (B, 1, 1)
+        context_embedded = has_text_context * context_text_embedded + (1 - has_text_context) * context_audio_embedded
+        context_lens = (
+            batch['has_text_context'].float() * context_text_lens
+            + (1 - batch['has_text_context'].float()) * context_audio_codes_lens
+        )  # (B,)
```

Comment:
It is not relevant to this PR, but why are the tensors interpolated in this way (multiplying by 0 or 1) instead of using e.g. `torch.where(has_text_context, text_context, audio_context)`

Reply:
Changed it to `torch.where`